DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/src/ac_int.cpp

```diff
@@ -0,0 +1,236 @@
+using MyInt28 = ac_int<28, true>;
+
+void TestBasicOpsInt(queue &q, const int &a, const int &b, int &c, int &d,
+                     int &e) {
+  buffer<int, 1> a_buf(&a, 1);
+  buffer<int, 1> b_buf(&b, 1);
+  buffer<int, 1> c_buf(&c, 1);
+  buffer<int, 1> d_buf(&d, 1);
+  buffer<int, 1> e_buf(&e, 1);
+
+  q.submit([&](handler &h) {
+    accessor a_acc(a_buf, h, read_only);
+    accessor b_acc(b_buf, h, read_only);
+    accessor c_acc(c_buf, h, write_only, no_init);
+    accessor d_acc(d_buf, h, write_only, no_init);
+    accessor e_acc(e_buf, h, write_only, no_init);
+    h.single_task<BasicOpsInt>([=]() [[intel::kernel_args_restrict]] {
+      c_acc[0] = a_acc[0] + b_acc[0];
+      d_acc[0] = a_acc[0] * b_acc[0];
+      e_acc[0] = a_acc[0] / b_acc[0];
+    });
+  });
+}
+
+void TestBasicOpsAcInt(queue &q, const MyInt14 &a, const MyInt14 &b, MyInt15 &c,
+                       MyInt28 &d, MyInt15 &e) {
+  buffer<MyInt14, 1> a_buf(&a, 1);
+  buffer<MyInt14, 1> b_buf(&b, 1);
+  buffer<MyInt15, 1> c_buf(&c, 1);
+  buffer<MyInt28, 1> d_buf(&d, 1);
+  buffer<MyInt15, 1> e_buf(&e, 1);
+
+  q.submit([&](handler &h) {
+    accessor a_acc(a_buf, h, read_only);
+    accessor b_acc(b_buf, h, read_only);
+    accessor c_acc(c_buf, h, write_only, no_init);
+    accessor d_acc(d_buf, h, write_only, no_init);
+    accessor e_acc(e_buf, h, write_only, no_init);
+    h.single_task<BasicOpsAcInt>([=]() [[intel::kernel_args_restrict]] {
+      c_acc[0] = a_acc[0] + b_acc[0];
+      d_acc[0] = a_acc[0] * b_acc[0];
+      e_acc[0] = a_acc[0] / b_acc[0];
+    });
+  });
+}
+
+void TestShiftOp(queue &q, const MyInt14 &a, const MyInt14 &b, MyInt14 &c) {
+  buffer<MyInt14, 1> a_buf(&a, 1);
+  buffer<MyInt14, 1> b_buf(&b, 1);
+  buffer<MyInt14, 1> c_buf(&c, 1);
+
+  q.submit([&](handler &h) {
+    accessor a_acc(a_buf, h, read_only);
+    accessor b_acc(b_buf, h, read_only);
+    accessor c_acc(c_buf, h, write_only, no_init);
+    h.single_task<ShiftOp>([=]() [[intel::kernel_args_restrict]] {
+      c_acc[0] = a_acc[0] << b_acc[0];
+    });
+  });
+}
+
+void TestEfficientShiftOp(queue &q, const MyInt14 &a, const MyUInt2 &b,
+                          MyInt14 &c) {
+  buffer<MyInt14, 1> a_buf(&a, 1);
+  buffer<MyUInt2, 1> b_buf(&b, 1);
+  buffer<MyInt14, 1> c_buf(&c, 1);
+
+  q.submit([&](handler &h) {
+    accessor a_acc(a_buf, h, read_only);
+    accessor b_acc(b_buf, h, read_only);
+    accessor c_acc(c_buf, h, write_only, no_init);
+    h.single_task<EfficientShiftOp>([=]() [[intel::kernel_args_restrict]] {
+      c_acc[0] = a_acc[0] << b_acc[0];
+    });
+  });
+}
+
+MyInt14 TestBitOps(queue &q, const MyInt14 &a) {
+  MyInt14 res;
+  buffer<MyInt14, 1> a_buf(&a, 1);
+  buffer<MyInt14, 1> res_buf(&res, 1);
+
+  q.submit([&](handler &h) {
+    accessor a_acc(a_buf, h, read_only);
+    accessor res_acc(res_buf, h, write_only, no_init);
+    h.single_task<BitOps>([=]() [[intel::kernel_args_restrict]] {
+      MyInt14 temp = a_acc[0].slc<14>(0);
```

Comment:
This is kind of a pointless case, why not slice to a different size? 
```suggestion
      MyInt7 temp = a_acc[0].slc<7>(3);
```